tools/nnyeah/NNyeahTests/TestRunning.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.IO;
+using System.Linq;
+using NUnit.Framework;
+
+namespace Microsoft.MaciOS.Nnyeah.Tests {
+	public class TestRunning {
+		static string GetInvokingTestName (out string callingMethodClass, string callingMethodName)
+		{
+			var stackTrace = new System.Diagnostics.StackTrace ();
+			var callingMethod = stackTrace.GetFrame (2)!.GetMethod ();
+			if (callingMethod is null)
+				Assert.Fail ("unable to get calling test from stack frame.");
+			if (string.IsNullOrEmpty (callingMethodName)) {
+				if (!callingMethod!.CustomAttributes.Any (x => x.AttributeType.Name == "TestAttribute")) {
+					Assert.Fail ("TestRunning expect invocations without an explicit `testName` parameter to be invoked from the [Test] method directly. Consider passing an explicit `testName`.");
+				}
+				callingMethodName = callingMethod.Name;
+			}
+			callingMethodClass = callingMethod!.DeclaringType!.Name;
+			return callingMethodName;
+		}
+
+		public static void TestAndExecute (string libraryCode, string callingCode, string expectedOutput,
```

Comment:
callingCode and expectedOutput are not used.

Reply:
Ditto. I'm just going to leave these ununsed bits in case we wire in test execution.